gocourse05/main.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import "fmt"
+
+type VisionType string
+
+const (
+	day   VisionType = "day"
+	night VisionType = "night"
+)
+
+type Event struct {
+	cameraID  int
+	eventType string
+	animal    string
+	action    string
+}
+
+type Camera interface {
+	TakeScreenShot(animal string, action string) Event
+}
+
+type DayCamera struct {
+	cameraId   int
+	visionType VisionType
+}
+
+func (c DayCamera) TakeScreenShot(animal string, action string) Event {
+	fmt.Printf("This is a %s camera registered a %s move for %s\n", c.visionType, action, animal)
+	return Event{
+		cameraID:  c.cameraId,
+		eventType: "Movement",
+		animal:    animal,
+		action:    action,
+	}
+}
+
+type NightCamera struct {
+	cameraId   int
+	visionType VisionType
+}
+
+func (c NightCamera) Describe() {
+	fmt.Printf("This is a %s camera %d\n", c.visionType, c.cameraId)
+}
+
+func (c NightCamera) TakeScreenShot(animal string, action string) Event {
+	fmt.Printf("This is a %s camera registered a %s move for %s\n", c.visionType, action, animal)
+	return Event{
+		cameraID:  c.cameraId,
+		eventType: "Movement",
+		animal:    animal,
+		action:    action,
+	}
+}
+
+type Zone struct {
+	name         string
+	dayCameras   map[int]*DayCamera
+	nightCameras map[int]*NightCamera
```

Comment:
Спробуй модифікувати код так, щоб працювала наступне:
```suggestion
	сameras map[int]Camera
```